### PR TITLE
Verify resource command does not conflict with plugin command

### PR DESCRIPTION
**Pull request by etsai-stripe**
### Reviewers
r? @
cc @stripe/developer-products

 ### Summary

Add a test case to verify newly updated/added resource command does not conflict with existing plugin commands

### CODE REVIEW

**Comment by etsai-stripe** [comment]
@suz-stripe this will not catch the reverse case though.. ie. new plugin command conflicts with an existing resource command. can we do something similar on the plugin repo?

File: pkg/cmd/resources.go [hunk 1]
```diff
@@ -2,17 +2,45 @@ package cmd
...
+}
+
+// Release is the type that holds release data for a specific build of a plugin
+type Release struct {
+	Arch    string
+	OS      string
+	Version string
+	Sum     string
+}
+
```
Comment (suz-stripe) [inline pkg/cmd/resources.go:43]: ~~Are we able to import from the plugins package in the `resources_test.go` file instead to make use of the same types defined in there? Then we can remove these. What do you think?~~

See my note about `getPluginCommands` instead 😃

File: pkg/cmd/resources.go [hunk 2]
```diff
@@ -2,17 +2,45 @@ package cmd
 
 import (
 	"fmt"
+	"io/ioutil"
+	"net/http"
 
+	"github.com/pelletier/go-toml"
```
Comment (suz-stripe) [inline pkg/cmd/resources.go:8]: did we switch toml packages recently? We normally use `github.com/BurntSushi/toml` IIRC

File: pkg/cmd/resources.go [hunk 3]
```diff
@@ -36,3 +64,30 @@ Use "stripe [command] --help" for more information about a command.
...
+		return []string{}, err
+	}
+
+	var pluginCommands []string
+	for _, plugin := range manifest.Plugins {
+		pluginCommands = append(pluginCommands, plugin.Shortname)
+	}
+
+	return pluginCommands, nil
+}
```
Comment (suz-stripe) [inline pkg/cmd/resources.go:93]: this feels like it belongs in the plugins package as a public function in `utilities.go`, and then `resources_test.go` can import and use this as a public function. This means we can also not worry about defining the plugin structs in this file nor the test file.

**Review by suz-stripe — COMMENTED** [review]
Left some feedback, mostly around location of the code outside of the test file :)

**Comment by suz-stripe** [comment]
> @suz-stripe this will not catch the reverse case though.. ie. new plugin command conflicts with an existing resource command. can we do something similar on the plugin repo?

I think this is a good idea if you can swing it!

File: pkg/cmd/resources.go (hunk 2 above)
Comment (etsai-stripe) [reply pkg/cmd/resources.go:8]: i will switch it back

**Review by suz-stripe — APPROVED** [review]

**Pull request merged**